uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/LocatorSearchSource.qml

```diff
@@ -0,0 +1,390 @@
+                                                             minScore: g.minScore,
+                                                             outputLanguageCode: g.outputLanguageCode,
+                                                             outputSpatialReference: g.outputSpatialReference,
+                                                             preferredSearchLocation: g.preferredSearchLocation,
+                                                             resultAttributeNames: g.resultAttributeNames,
+                                                             searchArea: g.searchArea
+                                                         });
+        }
+
+        function search_with_method(search_method, querySuggest, area, maxResults) {
+            // Reset the tempParent allowing for old geoElements and extents to be freed.
+            internal.tempParent = { children: [] };
+
+            const task = locatorSearchSource.locatorTask;
+            if (internal.currentSearchId) {
+                task.cancelTask(internal.currentSearchId);
+            }
+
+            const params = internal.copy_geocode_params();
+            if (area) {
+                params.preferredSearchLocation = null;
+                params.searchArea = area;
+            }
+            if (maxResults) {
+                params.maxResults = maxResults;
+            }
+            internal.currentSearchId = search_method(querySuggest, params);
+        }
+
+        property Component resultGeoElement: Component {
+            Graphic {
+                PictureMarkerSymbol {
+                    url: 'qrc:///esri.com/imports/Esri/ArcGISRuntime/Toolkit/images/map_pin.png'
+                }
+            }
+        }
+
+        property Connections locatorTaskConnections: Connections {
+            target: locatorSearchSource.locatorTask
+            function onLoadStatusChanged() {
+                const task = locatorSearchSource.locatorTask;
+                if (task.loadStatus === Enums.LoadStatusLoaded) {
+                    internal.initLocator();
+                }
+            }
+            function onGeocodeStatusChanged() {
+                const task = locatorSearchSource.locatorTask;
+
+                if (task.geocodeStatus === Enums.TaskStatusErrored) {
+                    locatorSearchSource.searchCompleted([]);
+                    return;
+                }
+
+                if (task.geocodeStatus === Enums.TaskStatusCompleted) {
+                    const results = [];
+                    for (let i = 0; i < task.geocodeResults.length; i++) {
+                        const g = task.geocodeResults[i];
+                        // Be aware that every call to `g.extent` returns an underlying new object each time.
+                        // Only refer to `extent` if you don't want garbage collection issues.
+                        const extent = g.extent;
```

Comment:
In this instance - if the extent object is added to the `ListModel`, then the object will be garbage collect at some arbitrary point, and will return `undefined` when accessed later from the ListModel.

So you have to avoid multiple calls to `g.extent`, because the `extent` added to the ListModel` will be garbage collected unless it is referenced elsewhere.

Reply:
I think this is a deficiency of the ListModel implementation. It is accessing objects, but those are somehow kept out of the reach of the garbage collector.